api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -3237,5 +3240,34 @@ public void mergePatients_shouldMoveAllObsWithSameHierarchy() throws Exception {
 		assertEquals(1, encounterService.getEncounter(57).getObsAtTopLevel(false).size());
 		assertEquals(2, encounterService.getEncounter(57).getObsAtTopLevel(true).size());
 	}
+	
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldThrowAnAPIExceptionIfAttributesIsNull() {
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage(NULL_ERROR_MESSAGE_FOR_ATTRIBUTES_NULL);
+		patientService.getDuplicatePatientsByAttributes(null);
+	}
+	
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldThrowAnAPIExceptionIfAttributesIsEmpty() {
+		expectedException.expect(APIException.class);
+		expectedException.expectMessage(NULL_ERROR_MESSAGE_FOR_ATTRIBUTES_NULL);
+		patientService.getDuplicatePatientsByAttributes(new ArrayList<>());
+	}
 
+	/**
+	 * @see PatientServiceImpl#getDuplicatePatientsByAttributes(List)
+	 * attributes on a Person or Patient object. similar to: [gender, givenName,
+	 *            middleName, familyName]
+	 */
+	@Test
+	public void getDuplicatePatientsByAttributes_shouldReturnPatients() {
+		List<PersonAttribute> activePersonAttributes = Context.getPatientService().getPatient(2).getActiveAttributes();
+		List<String> activeAttributes = new ArrayList<>();
+		for (PersonAttribute personAttribute : activePersonAttributes) {
+			activeAttributes.add(personAttribute.getValue());
+		}
+		List<Patient> receivedPatientList = patientService.getDuplicatePatientsByAttributes(activeAttributes);
+		assertNotNull(receivedPatientList);
```

Comment:
You need to check the count

Reply:
Where are the tests for the other methods mentioned in the ticket description?

Reply:
I have added the check count and other some tests.

Reply:
The test is more conclusive if you test with something that returns some hits